docs/reference/fields.txt

```diff
@@ -576,93 +578,341 @@ you have three options:
+- ``:binary`` => ``BSON::Binary``
+- ``:boolean`` => ``Mongoid::Boolean``
+- ``:date`` => ``Date``
+- ``:date_time`` => ``DateTime``
+- ``:float`` => ``Float``
+- ``:hash`` => ``Hash``
+- ``:integer`` => ``Integer``
+- ``:object_id`` => ``BSON::ObjectId``
+- ``:range`` => ``Range``
+- ``:regexp`` => ``Regexp``
+- ``:set`` => ``Set``
+- ``:string`` => ``String``
+- ``:stringified_symbol`` => ``StringifiedSymbol``
+- ``:symbol`` => ``Symbol``
+- ``:time`` => ``Time``
+
+
+.. _custom-id:
+
+Custom IDs
+==========
+
+By default, Mongoid defines the ``_id`` field on documents to contain a
+``BSON::ObjectId`` value which is automatically generated by Mongoid.
+
+It is possible to replace the ``_id`` field definition to change the type
+of the ``_id`` values or have different default values:
+
+.. code-block:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :name, type: String
+    field :_id, type: String, default: ->{ name }
+  end
+
+It is possible to omit the default entirely:
+
+.. code-block:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :_id, type: String
+  end
+
+If the default on ``_id`` is omitted, and no ``_id`` value is provided by
+your application, Mongoid will persist the document without the ``_id``
+value. In this case, if the document is a top-level document, an ``_id``
+value will be assigned by the server; if the document is an embedded document,
+no ``_id`` value will be assigned. Mongoid will not automatically retrieve
+this value, if assigned, when the document is persisted - you
+must obtain the persisted value (and the complete persisted document) using
+other means:
+
+.. code-block:: ruby
+
+  band = Band.create!
+  => #<Band _id: , >
+  band.id
+  => nil
+  band.reload
+  # raises Mongoid::Errors::DocumentNotFound
+  Band.last
+  => #<Band _id: 5fc681c22c97a6791f324b99, >
+
+Omitting ``_id`` fields is more common in :ref:`embedded documents <omit-id>`.
+
+Mongoid also defines the ``id`` field aliased to ``_id``. The ``id``
+alias can :ref:`be removed <unalias-id>` if desired (such as to integrate
+with systems that use the ``id`` field to store value different from ``_id``.
+
+
+.. _field-default-values:
+
+Default Values
+==============
+
+A field can be configured to have a default value. The default value can be
+fixed, as in the following example:
+
+.. code-block:: ruby
+
+  class Order
+    include Mongoid::Document
+
+    field :state, type: String, default: 'created'
+  end
+
+The default value can also be specified as a ``Proc``:
+
+.. code-block:: ruby
+
+  class Order
+    include Mongoid::Document
+
+    field :fulfill_by, type: Time, default: ->{ Time.now + 3.days }
+  end
+
+.. note::
+
+  Default values that are not ``Proc`` instances are evaluated at class load
+  time, meaning the following two definitions are not equivalent:
+
+  .. code-block:: ruby
+
+    field :submitted_at, type: Time, default: Time.now
+    field :submitted_at, type: Time, default: ->{ Time.now }
```

Comment:
Perhaps now would be the right time to mention that using associations in procs won't work, rather than above.